src/components/secure-backup/index.test.tsx

```diff
@@ -16,122 +18,250 @@ describe('SecureBackup', () => {
+        backupExists: true,
+        isBackupRecovered: true,
+        recoveryKey: '',
+      });
+
+      expect(wrapper).toHaveElement(RecoveredBackup);
     });
 
-    expect(wrapper).toHaveElement(Button);
+    it('publishes onClose', function () {
+      const onClose = jest.fn();
+      const wrapper = subject({
+        backupStage: BackupStage.None,
+        backupExists: true,
+        isBackupRecovered: true,
+        recoveryKey: '',
+        onClose,
+      });
+
+      wrapper.find(RecoveredBackup).prop('onClose')();
+
+      expect(onClose).toHaveBeenCalled();
+    });
+
+    it('publishes onGenerate when isLegacy is true', function () {
+      const onGenerate = jest.fn();
+      const wrapper = subject({
+        backupStage: BackupStage.None,
+        backupExists: true,
+        isBackupRecovered: true,
+        recoveryKey: '',
+        onGenerate,
+        isLegacy: true,
+      });
+
+      wrapper.find(RecoveredBackup).prop('onGenerate')();
+
+      expect(onGenerate).toHaveBeenCalled();
+    });
   });
 
-  it('renders success message', function () {
-    const wrapper = subject({ successMessage: 'It worked!' });
+  describe(GenerateBackup, () => {
+    it('renders GenerateBackup component when in GenerateBackup stage', function () {
+      const wrapper = subject({ backupStage: BackupStage.GenerateBackup });
+
+      expect(wrapper).toHaveElement(GenerateBackup);
+    });
+
+    it('publishes onSave', function () {
+      const onSave = jest.fn();
+      const wrapper = subject({ backupStage: BackupStage.GenerateBackup, onSave });
+
+      wrapper.find(GenerateBackup).prop('onSave')();
 
-    expect(wrapper.find(Alert).prop('variant')).toEqual('success');
-    expect(wrapper.find(Alert).childAt(0).text()).toEqual('It worked!');
+      expect(onSave).toHaveBeenCalled();
+    });
   });
 
-  it('renders error message', function () {
-    const wrapper = subject({ errorMessage: 'It failed!' });
+  describe(RestoreBackup, () => {
+    it('renders RestoreBackup component when in RestoreBackup stage', function () {
+      const wrapper = subject({ backupStage: BackupStage.RestoreBackup });
 
-    expect(wrapper.find(Alert).prop('variant')).toEqual('error');
-    expect(wrapper.find(Alert).childAt(0).text()).toEqual('It failed!');
+      expect(wrapper).toHaveElement(RestoreBackup);
+    });
+
+    it('publishes onRestore', function () {
+      const onRestore = jest.fn();
+      const wrapper = subject({ backupStage: BackupStage.RestoreBackup, onRestore });
+
+      wrapper.find(RestoreBackup).prop('onRestore')('abcd 1234');
+
+      expect(onRestore).toHaveBeenCalledWith('abcd 1234');
+    });
   });
-});
 
-function pressButton(wrapper, label: string) {
-  buttonLabelled(wrapper, label).simulate('press');
-}
+  describe(Success, () => {
+    it('renders Success component when in Success stage', function () {
+      const wrapper = subject({ backupStage: BackupStage.Success, successMessage: 'success' });
```

Comment:
Is the successMessage important to these tests? It seems like it's not so I'd remove it to make it clear which property is impacting the rendering decision.

Reply:
Not important, now removed.